File: output/PrimesFFICheatcode.php
<?php
$exports['runPrimesFFICheatcode'] = function($limit) {
    $n = (int)$limit;
    if ($n < 2) return 0;
    $sieve = array_fill(0, $n + 1, 1);
    
    for ($p = 2; $p * $p <= $n; $p++) {
        if ($sieve[$p]) {
            for ($i = $p * $p; $i <= $n; $i += $p) {
                $sieve[$i] = 0;
            }
        }
    }
    
    $sum = 0;
    for ($p = 2; $p <= $n; $p++) {
        if ($sieve[$p]) {
            $sum += $p;
        }
    }
    return $sum;
};
return $exports;

File: scratch/records-tast-opportunity-20260916/full-scalar/src/Test/PrimesFFICheatcode.erl
-module(test_primesFFICheatcode@foreign).
-export([runPrimesFFICheatcode/1]).
runPrimesFFICheatcode(N) -> primes(N, 2, []).
primes(Limit, Curr, Acc) when Curr > Limit -> sum(Acc, 0);
primes(Limit, Curr, Acc) ->
  case any_divides(Curr, Acc) of
    true -> primes(Limit, Curr + 1, Acc);
    false -> primes(Limit, Curr + 1, [Curr | Acc])
  end.
any_divides(_N, []) -> false;
any_divides(N, [X | Xs]) ->
  if N rem X == 0 -> true;
     true -> any_divides(N, Xs)
  end.
sum([], Acc) -> Acc;
sum([X | Xs], Acc) -> sum(Xs, Acc + X).
